microservices/authentication/src/queries/services.py

```diff
@@ -0,0 +1,56 @@
+import base64
+import time
+
+import jwt
+from dependency_injector.wiring import (
+    Provide,
+)
+from minos.common import (
+    AggregateDiff,
+)
+from minos.cqrs import (
+    QueryService,
+)
+from minos.networks import (
+    Request,
+    Response,
+    RestRequest,
+    enroute,
+)
+
+from ..jwt_env import (
+    JWT_ALGORITHM,
+    SECRET,
+)
+from .repositories import (
+    UserQueryRepository,
+)
+
+
+class LoginQueryService(QueryService):
+    repository: UserQueryRepository = Provide["user_repository"]
+
+    @enroute.rest.query("/login", "GET")
+    async def get_token(self, request: RestRequest) -> Response:
+        auth_type, encoded_credentials = request.raw_request.headers["Authorization"].split()
+        if auth_type == "Basic":
+            user, password = base64.b64decode(encoded_credentials).decode().split(":")
+
+            if await self.valid_credentials(user, password):
+                jwt_token = await self.generate_token(user)
+                return Response(jwt_token)
+            else:
+                return Response("Invalid username or password")
```

Comment:
Instead of returning a `Response`, you can raise a `ResponseException`.

Reply:
Yeah, I plan to refactor that in the future :smiley: